src/main/java/org/jabref/logic/journals/PredatoryJournalLoader.java

```diff
@@ -0,0 +1,140 @@
+            repository = new PredatoryJournalRepository(tempJournalList);
+            tempDir.toFile().deleteOnExit();
+            tempJournalList.toFile().deleteOnExit();
+        } catch (IOException e) {
+            LOGGER.error("Error while copying journal list", e);
+            return null;
+        }
+
+        // Update from external sources
+        if (doUpdate) {
+            update();
+        }
+
+        return repository;
+    }
+
+    private static void update() {
+        PREDATORY_SOURCES.forEach(PredatoryJournalLoader::crawl);       // populates linkElements (and predatoryJournals if CSV)
+        linkElements.forEach(PredatoryJournalLoader::clean);            // adds cleaned HTML to predatoryJournals
+
+        LOGGER.info("UPDATED PREDATORY JOURNAL LIST");
+    }
+
+    private static void crawl(PJSource source) {
+        try {
+            URLDownload download = new URLDownload(source.URL);
+
+            if (!download.canBeReached()) {
+                LOGGER.warn("URL UNREACHABLE");
+            } else if (source.URL.getPath().contains(".csv")) {
+                handleCSV(new InputStreamReader(download.asInputStream()));
+            } else {
+                handleHTML(source.ELEMENT_REGEX, download.asString());
+            }
+        } catch (IOException ex) {
+            logException(ex);
+        } catch (UnirestException ex) {
+            logException(ex);
+        }
+    }
+
+    private static void handleCSV(Reader reader) throws IOException {
+        CSVParser csvParser = new CSVParser(reader, CSVFormat.EXCEL);
+
+        for (CSVRecord csvRecord : csvParser) {
+            // changes column order from CSV (source: url, name, abbr)
+            repository.addToPredatoryJournals(csvRecord.get(1), csvRecord.get(2), csvRecord.get(0));
+        }
+    }
+
+    private static void handleHTML(String regex, String body) {
+        var pattern = Pattern.compile(regex);
+        var matcher = pattern.matcher(body);
+
+        while (matcher.find()) {
+            linkElements.add(matcher.group());
+        }
+    }
+
+    private static void clean(String item) {
+        var m_name = Pattern.compile("(?<=\">).*?(?=<)").matcher(item);
+        var m_url = Pattern.compile("http.*?(?=\")").matcher(item);
+        var m_abbr = Pattern.compile("(?<=\\()[^\s]*(?=\\))").matcher(item);
+
+        // using `if` gets only first link in element, `while` gets all, but this may not be desirable
+        // e.g. this way only the hijacked journals are recorded and not the authentic originals
+        if (m_name.find() && m_url.find()) {
+            repository.addToPredatoryJournals(m_name.group(), m_abbr.find() ? m_abbr.group() : "", m_url.group());
+        }
+    }
+
+    private static void logException(Exception ex) {
+        if (LOGGER.isErrorEnabled()) {
+            LOGGER.error(ex.getMessage(), ex);
+        }
+    }
```

Comment:
Please inline. `LOGGER.error` already checks whether it is enabled.

Loggers have to be

    LOGGER.error("text", ex);

`text` should be something in the context of the current method, not the ex.getMesasge